datafusion/expr/src/expr_schema.rs

```diff
@@ -136,7 +136,7 @@ impl ExprSchemable for Expr {
                         fun.return_type(&arg_data_types)
                     }
                     ScalarFunctionDefinition::UDF(fun) => {
-                        Ok(fun.return_type(&arg_data_types)?)
+                        fun.return_type_from_exprs(args, schema)
```

Comment:
@alamb @brayanjuls . It's here, when it will get the return type of `ScalarUDF`, and for UDF, it should be changed to this.
However, this can't work because `schema` needs to be `Sized`. Here is the error from my complier.
<img width="952" alt="Screenshot 2024-02-08 at 9 32 38 PM" src="https://github.com/apache/arrow-datafusion/assets/69566909/9cfdff48-0940-45d2-bddc-f0f60a40dfd0">